index.html

```diff
@@ -194,49 +198,227 @@ <h2>
+                <ol>
+                  <li>
+                    <p>For each media type <var>T</var> in
+                    <var>requestedMediaTypes</var>,</p>
+                    <ol>
+                      <li>
+                        <p>If no sources of type <var>T</var> are available,
+                        <a>reject</a> <var>p</var> with a new
+                        <code><a>DOMException</a></code> object whose
+                        <code><a>name</a></code> attribute has the value
+                        <code>NotFoundError</code>.</p>
+                      </li>
+                      <li>
+                        <p><a>Retrieve the permission state</a> for obtaining
+                        sources of type <var>T</var> in the current browsing
+                        context. If the permission state is "denied", jump to
+                        the step labeled <em>PermissionFailure</em> below.</p>
+                      </li>
+                    </ol>
+                  </li>
+                  <li>
+                    <p>Optionally, e.g., based on a previously-established
+                    user preference, for security reasons, or due to platform
+                    limitations, jump to the step labeled <em>Permission
+                      Failure</em> below.</p>
+                  </li>
+                  <li>
+                    <p>For the origin identified by
+                    <var>originIdentifier</var>, <a>prompt the user to choose</a>
+                    a display device, with a PermissionDescriptor with its name
+                    member set to the permission name associated with
+                    <var>kind</var> for obtaining display devices (e.g. "screen"
+                    for video), resulting in a set of provided media.</p>
+                    <p>The provided media MUST include precisely one track of
+                    each media type in <var>requestedMediaTypes</var>.
+                    The devices chosen MUST be the ones determined by the user.
+                    Once selected, the source of a
+                    <code><a>MediaStreamTrack</a></code> MUST NOT change.</p>
+                    <p>User Agents are encouraged to warn users against sharing
+                    <a>browser</a> display devices, or otherwise try to
+                    steer users away from sharing them.</p>
+                    <p>If the result of the request is "granted", then for
+                    each device that is sourcing the provided media, using
+                    the device's deviceId, <var>deviceId</var>, set
+                    [[\devicesLiveMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already <code>true</code>,
+                    and set the
+                    [[\devicesAccessibleMap]]<var>[deviceId]</var> to
+                    <code>true</code>, if it isn’t already
+                    <code>true</code>.</p>
```

Comment:
This per-device permission stuff—which no-one implements btw—has already received (legitimate) flak for being overly complicated, so I worry about adding complexity to this. Since the deviceIds are never exposed, can we perhaps use the raw internal ids which are stable?

Reply:
If the point here is to support indicators, then sure.

Reply:
I think this needs clarification too.